src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/RestrictedPatientsService.kt

```diff
@@ -99,31 +99,36 @@ class RestrictedPatientsService(
     val prisonerResult = prisonerSearchResponse.firstOrNull()
       ?: throw NoResultsReturnedException("No prisoner search results returned for $prisonerNumber")
 
-    prisonApiGateway.createExternalMovement(
-      CreateExternalMovement(
-        bookingId = prisonerResult.bookingId,
-        fromAgencyId = restrictedPatient.hospitalLocationCode,
-        toAgencyId = "OUT",
-        movementTime = LocalDateTime.now(clock),
-        movementType = "REL",
-        movementReason = "CR",
-        directionCode = "OUT"
-      )
-    )
-
     restrictedPatientsRepository.delete(restrictedPatient)
 
-    telemetryClient.trackEvent(
-      "restricted-patient-removed",
-      mapOf(
-        "prisonerNumber" to prisonerNumber,
-        "fromLocationId" to restrictedPatient.fromLocationId,
-        "hospitalLocationCode" to restrictedPatient.hospitalLocationCode,
-        "supportingPrisonId" to restrictedPatient.supportingPrisonId,
-        "dischargeTime" to restrictedPatient.dischargeTime.toString(),
-      ),
-      null
-    )
+    try {
+      prisonApiGateway.createExternalMovement(
+        CreateExternalMovement(
+          bookingId = prisonerResult.bookingId,
+          fromAgencyId = restrictedPatient.hospitalLocationCode,
+          toAgencyId = "OUT",
+          movementTime = LocalDateTime.now(clock),
+          movementType = "REL",
+          movementReason = "CR",
+          directionCode = "OUT"
+        )
+      )
+
+      telemetryClient.trackEvent(
+        "restricted-patient-removed",
+        mapOf(
+          "prisonerNumber" to prisonerNumber,
+          "fromLocationId" to restrictedPatient.fromLocationId,
+          "hospitalLocationCode" to restrictedPatient.hospitalLocationCode,
+          "supportingPrisonId" to restrictedPatient.supportingPrisonId,
+          "dischargeTime" to restrictedPatient.dischargeTime.toString(),
+        ),
+        null
+      )
+    } catch (e: Exception) {
+      restrictedPatientsRepository.saveAndFlush(restrictedPatient)
```

Comment:
This creates a copy of the original restricted patient record with an incremented id (PK). In future PR's we're planning on making prisoner number the primary key as we're already treating it as one.